source/kernel/src/services/sdmmc.rs

```diff
@@ -0,0 +1,489 @@
+        match self
+            .cmd(Command {
+                index: 2,
+                argument: 0,
+                options: HardwareOptions::None,
+                kind: CommandKind::Control,
+                rsp_type: ResponseType::Long,
+                rsp_crc: true,
+                buffer: None,
+            })
+            .await?
+        {
+            Response::Short { .. } => Err(Error::Response),
+            Response::Long(value) => {
+                tracing::trace!("CMD2 response: {value:?}");
+                union CidResponse {
+                    value: [u32; 4],
+                    cid: u128,
+                }
+                let rsp = CidResponse { value };
+                Ok(CardIdentification(unsafe { rsp.cid }))
+            }
+        }
+    }
+
+    /// Get the relative card address
+    pub async fn get_rca(&mut self) -> Result<RelativeCardAddress, Error> {
+        match self
+            .cmd(Command {
+                index: 3,
+                argument: 0,
+                options: HardwareOptions::None,
+                kind: CommandKind::Control,
+                rsp_type: ResponseType::Short,
+                rsp_crc: true,
+                buffer: None,
+            })
+            .await?
+        {
+            Response::Short { value, .. } => {
+                tracing::trace!("CMD3 response: {value:#x}");
+                Ok(RelativeCardAddress(value))
+            }
+            Response::Long(_) => Err(Error::Response),
+        }
+    }
+
+    /// Toggle the card between stand-by and transfer state
+    pub async fn select(&mut self, rca: RelativeCardAddress) -> Result<CardStatus, Error> {
+        match self
+            .cmd(Command {
+                index: 7,
+                argument: rca.0,
+                options: HardwareOptions::None,
+                kind: CommandKind::Control,
+                rsp_type: ResponseType::Short,
+                rsp_crc: true,
+                buffer: None,
+            })
+            .await?
+        {
+            Response::Short { value, .. } => {
+                tracing::trace!("CMD7 response: {value:#x}");
+                Ok(CardStatus(value))
+            }
+            Response::Long(_) => Err(Error::Response),
+        }
+    }
+
+    /// Use 4 data lanes
+    pub async fn set_wide_bus(&mut self, rca: RelativeCardAddress) -> Result<CardStatus, Error> {
+        // Go to *APP* mode before sending application command
+        let _ = self
+            .cmd(Command {
+                index: 55,
+                argument: rca.0,
+                options: HardwareOptions::None,
+                kind: CommandKind::Control,
+                rsp_type: ResponseType::Short,
+                rsp_crc: true,
+                buffer: None,
+            })
+            .await?;
+
+        match self
+            .cmd(Command {
+                index: 6,
+                argument: 0b10,
```

Comment:
what does 0b10 mean here?

Reply:
For this command, you can use either `0b00` as argument, to tell the card to switch to 1-bit width or `0b10` to switch to 4-bit width. I added a comment to clarify.
I guess this `set_wide_bus` function should have been `set_bus_width` though, with an argument that specifies the width.
If it's okay for you, I'd prefer to postpone this change to a future PR though.